packages/ts/react-crud/src/autogrid-column-context.tsx

```diff
@@ -10,12 +11,18 @@ export interface SorterState {
 
 export type SortState = Record<string, SorterState | undefined>;
 
-export type ColumnContext = {
+export type ColumnContext = Readonly<{
   propertyInfo: PropertyInfo;
   setPropertyFilter(propertyFilter: PropertyStringFilter): void;
   sortState: SortState;
   setSortState: Dispatch<SetStateAction<SortState>>;
   customColumnOptions?: ColumnOptions;
-};
+}>;
 
 export const ColumnContext = createContext<ColumnContext | null>(null);
+
+export type GeneralColumnContext = Readonly<{
```

Comment:
How about using a context that can be wrapped around the whole grid instead? The default context value can have a formatter with an `undefined` locale, which uses the browser locale by default. For the tests, we can explicitly wrap the grid into a context using a fixed locale.

This would also avoid adding public API at this point, at least in a sense that the context is not as discoverable as a prop.

Reply:
Yeah, actually, this context wraps the whole grid. Maybe, the naming is not good enough...

Reply:
Sorry, what I meant was: Remove the `locale` prop from auto grid, and instead wrap a context around the auto grid to provide either a formatter or a locale.

Reply:
Hmm... I was sure that React will throw an error if there is no `Provider` for the context but after you mentioned it I decided to check. And it looks like the `Provider` is optional 🤔. Let me try that option then. Sounds pretty interesting.

Reply:
If you don't use an explicit provider, then accessing the context will give you the default value that you need to specify when creating the context.

Reply:
@sissbruecker thanks, works like a charm!